lib/zstd.h

```diff
@@ -1072,6 +1072,14 @@ ZSTDLIB_API size_t ZSTD_sizeof_DDict(const ZSTD_DDict* ddict);
 
 typedef struct ZSTD_CCtx_params_s ZSTD_CCtx_params;
 
+typedef struct {
```

Comment:
We probably need to document that structure, so that any user can quickly understand what it's about,
and the fields' meaning is clearly established.
For example, I'm not sure just from the naming if `matchPos` is the position of the match at its destination point (I presume), or at its source point. Also it's relative to something (likely `src`).
There was also this discussion about the meaning of `rep` values.
Just clarify, for any 3rd party.